files/scripts/sonic-netns-exec

```diff
@@ -6,7 +6,7 @@
 NS="$1"
 shift
 if [ "$NS" != "" ]; then
-    ip netns exec $NS $@
+    ip netns exec $NS "$@"
 else
-    $@
+    "$@"
```

Comment:
should be -> eval $@

Reply:
on line 8, to check if a string is empty or not, should use the standard way. if [ -z "$var" ]

Reply:
check below. I am not sure adding quote is the correct solution.

```
lgh@gulv-vm3:/data2/sonic/stretch/sonic-buildimage$ "ip addr show"
ip addr show: command not found
```

Reply:
Updated condition to check if string is empty or not. Updated description to explain the main issue seen and fix.

Reply:
I have updated the description and title to reflect the fix.

Reply:
https://wiki.bash-hackers.org/scripting/posparams


Syntax | Effective result
-- | --
$* | $1 $2 $3 … ${N}
$@ | $1 $2 $3 … ${N}
"$*" | "$1c$2c$3c…c${N}"
"$@" | "$1" "$2" "$3" … "${N}"

Without being quoted (double quotes), both have the same effect: All positional parameters from $1 to the last one used are expanded without any special handling.

When the $* special parameter is double quoted, it expands to the equivalent of: "$1c$2c$3c$4c……..$N", where 'c' is the first character of IFS.

But when the $@ special parameter is used inside double quotes, it expands to the equivanent of…

"$1" "$2" "$3" "$4" ….. "$N"

…which reflects all positional parameters as they were set initially and passed to the script or function. If you want to re-use your positional parameters to call another program (for example in a wrapper-script), then this is the choice for you, use double quoted "$@".

Well, let's just say: You almost always want a quoted "$@"!